client/test/e2e/patient_invoice/modal.page.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * Created by Dedrick Kitamuka on 15/06/2016.
+ */
+
+function modalPage () {
+    var page = this;
+
+    var todayButton = element(by.id('today'));
+    var weekButton = element(by.id('week'));
+    var monthButton = element(by.id('month'));
+    var yearButton = element(by.id('year'));
+
+    var referenceField = element(by.id('reference'));
+
+    var serviceSelect = element(by.id('service'));
+    var userSelect = element(by.id('user'));
+
+    var allRadio = element(by.id('all'));
+    var yesRadio = element(by.id('yes'));
+    var noRadio = element(by.id('no'));
+
+
+    var submitButton = element(by.id('submitButton'));
+
+
+    //interfaces
+    function chooseToday() {
+        todayButton.click();
+    }
+
+    function chooseWeek (){
+        weekButton.click();
+    }
+
+    function chooseMonth (){
+        monthButton.click();
+    }
+
+    function chooseYear (){
+        yearButton.click();
+    }
+
+    function setReference(value) {
+        referenceField.clear().sendKeys(value);
+    }
+    
+    function setServiceChoice (choice) {
+        serviceSelect.element(by.cssContainingText('option', choice));
+    }
+
+    function setUserChoice (choice) {
```

Comment:
For these `<select>` elements, you could use `FormUtilities` instead of re implementing this functionality.  It would be something like
```js
FU.select('ModalCtrl.invoice.service_id', choice);
```